test/lib/jdk/test/lib/jvmti/jvmti_common.hpp

```diff
@@ -447,7 +447,25 @@ static jthread get_current_thread(jvmtiEnv *jvmti, JNIEnv* jni) {
   return thread;
 }
 
+/* Used in a couple of nsk/jvmti/scenarios tests to convert jbyteArray to a JVMTI allocated */
+static unsigned char* jni_array_to_jvmti_allocated(jvmtiEnv *jvmti, JNIEnv *jni, jbyteArray arr, jint* len_ptr) {
+    unsigned char* new_arr = nullptr;
 
+    jint len = jni->GetArrayLength(arr);
+    if (len <= 0) {
+      fatal(jni, "JNI GetArrayLength returned a non-positive value");
+    }
+    jbyte* jni_arr = jni->GetByteArrayElements(arr, nullptr);
```

Comment:
I think you need to release array before exit from this function to unpin it.

Reply:
Good catch, thanks. Fixed now.